test/realm_logger_test.dart

```diff
@@ -0,0 +1,337 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// ignore_for_file: unused_local_variable, avoid_relative_lib_imports
+
+import 'dart:async';
+import 'dart:isolate';
+import 'package:logging/logging.dart';
+import 'package:test/test.dart' hide test, throws;
+import '../lib/src/realm_class.dart' show RealmInternal;
+import '../lib/realm.dart';
+import 'test.dart';
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  test('Realm.logger expected messages for log levels', () async {
+    String generatedMessage = generateRandomString(10);
+    Map<Level, int> logMessages = await Isolate.run(() async {
+      Realm.logger = Logger.detached(generateRandomString(5))..level = RealmLogLevel.all;
+      final allMessages = await _attachToLoggerBeforeAction("root isolate", () => _simulateLogs(generatedMessage));
+      return _matchedMessagesCountPerLevel(allMessages, generatedMessage);
+    });
+    _expectAllLevelLogs(logMessages);
+  });
+
+  test('Realm.logger expected messages after action completed', () async {
+    String generatedMessage = generateRandomString(10);
+    Map<Level, int> logMessages = await Isolate.run(() async {
+      Realm.logger = Logger.detached(generateRandomString(5))..level = RealmLogLevel.all;
+      final allMessages = await _attachToLoggerAfterAction("root isolate", () => _simulateLogs(generatedMessage));
+      return _matchedMessagesCountPerLevel(allMessages, generatedMessage);
+    });
+    _expectAllLevelLogs(logMessages);
+  });
+
+  baasTest('Realm.logger level changed', (configuration) async {
+    Map<String, Map<Level, int>> logMessages = await Isolate.run(() async {
+      Map<String, Map<Level, int>> results = {};
```

Comment:
```suggestion
      final results = <String, Map<Level, int>>{};
```

I see we are using explicit types everywhere could we use `final` or `var` instead

Reply:
The test are harder to read cause it is too verbose on many places